examples/C++/Keys/keys.cpp

```diff
@@ -236,9 +253,10 @@ void keys()
     Subscriber* mySub = initSubscriber(sampleType, nullptr);
 
     // wait for the connection
-    while (pubListener.n_matched == 0)
+    if (pubListener.n_matched == 0)
     {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        std::unique_lock<std::mutex> lock(pubListener.mutex_);
+        pubListener.cv_.wait(lock);
```

Comment:
You can avoid the if by providing the CV a predicate: 

    pubListener.cv_.wait(lock, [&pubListener]()
        {
            return pubListener.n_matched > 0;
        });